engine/protocol/api.go

```diff
@@ -0,0 +1,142 @@
+package protocol
+
+import (
+	"context"
+	"errors"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type API interface {
```

Comment:
And same for the file name, maybe `protocol_api.go` versus `api.go`

Reply:
it will be referred using package_name.interface_name when it is imported, would be imported as `protocol.API` and interface name ProtocolAPI will have duplicate name like protocol.ProtocolAPI

Reply:
Ah ok, that makes sense. Thanks.